src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,476 @@
+                throw new ArgumentNullException(nameof(externalComObject));
+
+            return obj!;
+        }
+
+        /// <summary>
+        /// Get the currently registered managed object or creates a new managed object and registers it.
+        /// </summary>
+        /// <param name="impl">The <see cref="ComWrappers" /> implementation to use when creating the managed object.</param>
+        /// <param name="externalComObject">Object to import for usage into the .NET runtime.</param>
+        /// <param name="innerMaybe">The inner instance if aggregation is involved</param>
+        /// <param name="flags">Flags used to describe the external object.</param>
+        /// <param name="wrapperMaybe">The <see cref="object"/> to be used as the wrapper for the external object.</param>
+        /// <param name="retValue">The managed object associated with the supplied external COM object or <c>null</c> if it could not be created.</param>
+        /// <returns>Returns <c>true</c> if a managed object could be retrieved/created, <c>false</c> otherwise</returns>
+        /// <remarks>
+        /// If <paramref name="impl" /> is <c>null</c>, the global instance (if registered) will be used.
+        /// </remarks>
+        private static bool TryGetOrCreateObjectForComInstanceInternal(
+            ComWrappers impl,
+            IntPtr externalComObject,
+            IntPtr innerMaybe,
+            CreateObjectFlags flags,
+            object? wrapperMaybe,
+            out object? retValue)
+        {
+            if (externalComObject == IntPtr.Zero)
+                throw new ArgumentNullException(nameof(externalComObject));
+
+            if (flags.HasFlag(CreateObjectFlags.Aggregation))
+                throw new NotImplementedException();
+
+            // If the inner is supplied the Aggregation flag should be set.
+            // if (innerMaybe != IntPtr.Zero && !flags.HasFlag(CreateObjectFlags.Aggregation))
+            //    throw new InvalidOperationException(SR.InvalidOperation_SuppliedInnerMustBeMarkedAggregation);
+
+            object? wrapperMaybeLocal = wrapperMaybe;
+            retValue = null;
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Register a <see cref="ComWrappers" /> instance to be used as the global instance for reference tracker support.
+        /// </summary>
+        /// <param name="instance">Instance to register</param>
+        /// <remarks>
+        /// This function can only be called a single time. Subsequent calls to this function will result
+        /// in a <see cref="System.InvalidOperationException"/> being thrown.
+        ///
+        /// Scenarios where this global instance may be used are:
+        ///  * Object tracking via the <see cref="CreateComInterfaceFlags.TrackerSupport" /> and <see cref="CreateObjectFlags.TrackerObject" /> flags.
+        /// </remarks>
+        public static void RegisterForTrackerSupport(ComWrappers instance)
```

Comment:
It may be better for this to keep throwing not supported exception since it is 100% not implemented. You can keep the code, but put it under `#if false`